Updated hpc-stack modules and compiler flags for building SRW release/public-v2 on MacOS

natalie-perlin opened the pull request:
## DESCRIPTION OF CHANGES: 
Changed a build template for the MacOS system to use the hpc-stack modules built for the release/public-v2 SRW version, a file `./modulefiles/build_macos_gnu` .
Recipe for building the hpc-stack for this release on a MacOS using the GNU compiler suite, for either M1/arm64 or Intel/x86_64 chips could be found in HPC-stack repo:
https://github.com/NOAA-EPIC/hpc-stack.git, release/public-v2 branch; it will be later merged with the hpc-stack NOAA-EMC repository.  Currently it could be cloned as following:
`git clone -b release/public-v2 https://github.com/NOAA-EPIC/hpc-stack.git hpc-stack.git`

## TESTS CONDUCTED: 
The SRW was build and tested (single test 2019051600) based on the hpc-stack for release/public-v2. Tho MacOS systems tested: M1/arm64 with Darwin 20/Big Sure, Intel/x86_64 with Darwin 19/Catalina, build has been also tested on M1/arm64 with Darwin 21/Monterey.

## DEPENDENCIES:
HPC-stack recipe built for this release:
- NOAA-EPIC/hpc-stack/release/public-v2
- NOAA-EMC/hpc-stack/pull/250

## DOCUMENTATION:
The documentation is under a separate PR

## ISSUE (optional): 
1. This resolves dependence of syncing the modules loaded to build and run the SRW release/public-v2 on MacOS.
2. Modulefiles build_macos_gnu, wflow_macos could be loaded. Env. variable LDFLAGS needs to be set up (exported) after loading build_macos_gnu; wflow_macos needs to have "puts" in place of "system" calls.
  
  See discussion under the [PR #238 ](https://github.com/ufs-community/ufs-srweather-app/pull/238)
  
  Contributors:
  @danielabdi-noaa

Commit:
Updated versions of hpc-stack modules and compiler flags for building SRW release/public-v2

JeffBeck-NOAA commented:
@natalie-perlin, can you change the title of this PR to reflect that these changes are only for MacOS?  Thanks!

JeffBeck-NOAA commented:
@willmayfield, does this look good to you?

Commit:
Fix artifact creation for branches with / in name (#275)

The slash in branch names such as release/.* and feature/.* is
interpreted as a directory separator. This change replaces the "/"
character with a "_".

natalie-perlin committed:
Merge branch 'ufs-community:release/public-v2' into release/public-v2

natalie-perlin committed:
Corrections to the modulefiles for the MacOS

Modulefiles load software libraries and activate python virtual environment

natalie-perlin committed:
resolved conflict in modulefiles/build_macos_gnu

natalie-perlin committed:
correction in build_macos_gnu

natalie-perlin committed:
correction in build_macos_gnu

danielabdi-noaa commented:
@natalie-perlin This looks good to me. Sorry that the only way to get LDFLAGS working in `build_macos_gnu` seems to be if it was written in Lua according to [this document](https://lmod.readthedocs.io/en/latest/095_tcl2lua.html). Maybe printing a message  at the end of `build_macos_gnu` will do for now, but it is not an ideal solution either.
```csh
set shell [module-info shelltype]
if {$shell == "sh"} {
   puts stderr {Please execute this command:
        > export LDFLAGS=-L$MPI_ROOT/lib}
} else {
   puts stderr {Please execute this command:
        > setenv LDFLAGS -L$MPI_ROOT/lib}
}
```
